### ISSUE TITLE
Support complex types when using clients

### ISSUE THREAD

**Issue by moul** [gnolang/gno#925]
## Context

Clients only support passing and receiving primitive types.
Returned value is hard to parse.

We need to find a solution to support rich types. The current best approach is potentially to generate protobuf files, cue files, amino files dynamically.

## Examples of clients
- gnokey: https://github.com/gnolang/gno/tree/master/gno.land/cmd/gnokey
- gnofaucet: https://github.com/gnolang/gno/tree/master/gno.land/cmd/gnofaucet
- gnoweb: https://github.com/gnolang/gno/tree/master/gno.land/cmd/gnoweb
- adena-wallet: https://github.com/onbloc/adena-wallet
- gnoblog-cli: https://github.com/gnolang/blog/tree/main/gnoblog-cli

## Needed by
- https://github.com/gnolang/gno-js-client
- Teritori, returning complex types

**Comment by thehowl** [comment]
what about making a [gnob](https://pkg.go.dev/encoding/gob)? :)

could probably keep the same specification, but with the possibility of changing it based on how gno and its primitives evolve

**Comment by thehowl** [comment]
This is partly achieved with `gnokey maketx run`.

**Comment by zivkovicmilos** [comment]
Closing in favor of #1842
